internal/files/files.go

```diff
@@ -71,3 +72,28 @@ func FileType(fs vfs.FileSystem, path string) (string, error) {
 	// content-type by returning "application/octet-stream" if no others seemed to match.
 	return http.DetectContentType(buf), nil
 }
+
+// FindDirectoryContaining returns the path to the directory containing the targetFolderName
+// and a bool indicating if the directory was found or not.
+// If the directory was not found, the error will be os.ErrNotExist.
+// The search starts at the path and continues to the root of the filesystem.
+// The targetFolderName can be a relative path.
+// Example:
+//
+//	FindDirectoryContaining("` + path + `", "` + targetFolderName + `")
+func FindDirectoryContaining(path, targetFolderName string) (string, bool, error) {
```

Comment:
The error can already indicates if the directory exists or not. Seems the bool return value is redundant?